src/vm/memory/segments.zig

```diff
@@ -797,6 +826,46 @@ test "MemorySegmentManager: getSegmentSize should return null if missing segment
     try expectEqual(@as(?u32, null), memory_segment_manager.getSegmentSize(3));
 }
 
+test "MemorySegmentManager: relocateSegments for one segment" {
+    var memory_segment_manager = try MemorySegmentManager.init(std.testing.allocator);
+    defer memory_segment_manager.deinit();
+    try memory_segment_manager.segment_used_sizes.put(0, 1);
+    const actual_value = try memory_segment_manager.relocateSegments(std.heap.page_allocator);
+    var expected_value = ArrayList(usize).init(std.testing.allocator);
+    defer expected_value.deinit();
+    try expected_value.append(1);
+    try expectEqualSlices(usize, expected_value.items, actual_value);
+}
+
+test "MemorySegmentManager: relocateSegments for ten segments" {
+    var memory_segment_manager = try MemorySegmentManager.init(std.testing.allocator);
+    defer memory_segment_manager.deinit();
+    try memory_segment_manager.segment_used_sizes.put(0, 3);
+    try memory_segment_manager.segment_used_sizes.put(1, 7);
+    try memory_segment_manager.segment_used_sizes.put(2, 12);
+    try memory_segment_manager.segment_used_sizes.put(3, 15);
+    try memory_segment_manager.segment_used_sizes.put(4, 10);
+    try memory_segment_manager.segment_used_sizes.put(5, 17);
+    try memory_segment_manager.segment_used_sizes.put(6, 3);
+    try memory_segment_manager.segment_used_sizes.put(7, 30);
+    try memory_segment_manager.segment_used_sizes.put(8, 55);
+    try memory_segment_manager.segment_used_sizes.put(9, 60);
+    const actual_value = try memory_segment_manager.relocateSegments(std.heap.page_allocator);
```

Comment:
In tests, we should always use `std.testing.allocator`

Reply:
Same here